.github/workflows/release.yml

```
@@ -3,7 +3,7 @@ name: Release
 on:
   push:
     tags:
-      - "*"
+      - "[0-9]+.[0-9]+.[0-9]+"
```

This is a guardrail against a non-version tag being accidentally pushed and published to npm.

---

I don't think this will work as it also matches `1970-01-01` as an example. You may want something like this:

```suggestion
      - "^[0-9]+\.[0-9]+\.[0-9]+$"
```

`^` matches the beginning of the string, then we escape `.` to match the character literal, and `$` matches the end of the string. 

Without the `^` and `$`, something like `1.1.1.1.1.1.1.1.1.1.1` would match multiple capture groups.

---

The syntax for filtering tags and branches looks regexy, but it's actually got its [own filtering syntax](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet)! An example semver pattern is provided in that cheatsheet.

Apparently regex filters isn't something that is supported in workflows. [This thread](https://github.community/t/using-regex-for-filtering/16427/2) is a little old, but I haven't found anything more recent that indicates to the contrary.